library/core/src/unicode/mod.rs

```diff
@@ -31,3 +31,55 @@ mod unicode_data;
 /// [Unicode 11.0 or later, Section 3.1 Versions of the Unicode Standard](https://www.unicode.org/versions/Unicode11.0.0/ch03.pdf#page=4).
 #[stable(feature = "unicode_version", since = "1.45.0")]
 pub const UNICODE_VERSION: (u8, u8, u8) = unicode_data::UNICODE_VERSION;
+
+#[cfg(kani)]
+mod verify {
+    use super::conversions::{to_upper, to_lower};
+    use wrappers::*;
+    use crate::kani;
+    use safety::*;
+
+    /// Wrapper functions used to verify auto-generated functions from this module.
+    ///
+    /// The files in this module are auto-generated by a script, so they are harder to annotate.
+    /// Instead, for each function that we want to verify, we create a wrapper function with
+    /// contracts.
+    mod wrappers {
+        use super::*;
+        use crate::ub_checks;
+
+        /// Wraps `conversions::to_upper` function.
+        ///
+        /// ```no_run
+        /// pub fn to_upper(c: char) -> [char; 3] {
+        /// # todo!()
+        /// }
+        /// ```
+        #[ensures(|res| ub_checks::can_dereference(res))]
```

Comment:
Can this fail given that the return type is an array?

Reply:
Yes... that's exactly the test that I have recently added to kani (https://github.com/model-checking/kani/commit/041beda7afb6114a831de61a2ae68b0e892919b9). The `can_dereference` method today already checks the value validity (and it should eventually check initialization and aliasing rules).

Reply:
Fair. I'll remove the ensures clause and use a regular harness instead.